File: daily/Arrays/905.Sort-Array-By-Parity.py
from typing import List


class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        if len(nums) <= 1:
            return nums
        left, right = 0, len(nums) - 1
        while left < right:
            if nums[left] % 2 == 0:
                left += 1
            elif nums[right] % 2 != 0:
                right -= 1
            else:
                nums[left], nums[right] = nums[right], nums[left]
                left += 1
                right -= 1
        return nums

File: daily/Arrays/905.Sort-Array-By-Parity.go
package daily

func sortArrayByParity(nums []int) []int {
	if len(nums) == 1 {
		return nums
	}

	left, right := 0, len(nums)-1
	for left < right {
		if nums[left]%2 == 0 {
			left++
		} else if nums[right]%2 != 0 {
			right--
		} else {
			nums[left], nums[right] = nums[right], nums[left]
			left++
			right--
		}
	}

	return nums
}
